mountpoint-s3/tests/fs.rs

```diff
@@ -445,3 +445,46 @@ async fn test_stat_block_size() {
     assert_eq!(lookup.attr.blocks, 9);
     assert_eq!(lookup.attr.blksize, 4096);
 }
+
+#[test_case(""; "unprefixed")]
+#[test_case("test_prefix/"; "prefixed")]
+#[tokio::test]
+async fn test_local_dir(prefix: &str) {
+    let prefix = Prefix::new(prefix).expect("valid prefix");
+    let (client, fs) = make_test_filesystem("test_local_dir", &prefix, Default::default());
+
+    // Create local directory
+    let dirname = "local";
+    let dir_entry = fs
+        .mkdir(FUSE_ROOT_INODE, dirname.as_ref(), libc::S_IFDIR, 0)
+        .await
+        .unwrap();
+
+    assert_eq!(dir_entry.attr.kind, FileType::Directory);
+    let dir_ino = dir_entry.attr.ino;
+
+    assert!(!client.contains_prefix(&format!("{prefix}{dirname}")));
+
+    let lookup_entry = fs.lookup(FUSE_ROOT_INODE, dirname.as_ref()).await.unwrap();
+    assert_eq!(lookup_entry.attr, dir_entry.attr);
+
+    // Write an object into the directory
+    let filename = "file.bin";
+    let mode = libc::S_IFREG | libc::S_IRWXU; // regular file + 0700 permissions
+    let file_entry = fs.mknod(dir_ino, filename.as_ref(), mode, 0, 0).await.unwrap();
+    let file_ino = file_entry.attr.ino;
+    let file_handle = fs
+        .open(file_ino, libc::S_IFREG as i32 | libc::O_WRONLY)
+        .await
+        .unwrap()
+        .fh;
+
+    fs.release(file_ino, file_handle, 0, None, false).await.unwrap();
+
+    // Remove the new object from the client
+    client.remove_object(&format!("{prefix}{dirname}/{filename}"));
+
+    // Verify that the directory disappeared
+    let lookup = fs.lookup(FUSE_ROOT_INODE, dirname.as_ref()).await;
+    assert!(matches!(lookup, Err(libc::ENOENT)));
```

Comment:
This behavior should be documented somewhere, maybe in semantics doc.

Reply:
I'm adding a section for `mkdir` to the semantics doc. Should I add it to this PR or a new one?

Reply:
Either is fine.